internal/server/server.go

```diff
@@ -295,3 +318,58 @@ func (s *Server) handleCallback() http.HandlerFunc {
 		}
 	}
 }
+
+func (s *Server) handleHealthCheck() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		oidcResp, err := s.client.Get(s.issuerURL.String() + "/healthz")
+		if err != nil {
+			status := fmt.Sprintf("cannot connect to oidc provider. error: %v", err)
+			log.WithFields(log.Fields{
+				"issuerURL": s.issuerURL,
+				"err":       err,
+			}).Error(status)
+
+			writeJsonResponse(w, http.StatusBadGateway, healthCheckResponse{Status: status})
```

Comment:
We should not write open-ended error information back to the client.  

`cannot connect to "issuerURL"` is safe.   The `err` should only be logged.

Reply:
Fixed, here and all errors below